Ks network analysis report 1 (#213)
tari-labs/tari-university: src/network-analysis/probabilistic-attack/probabilistic_attack_vector_analysis_building_blocks.md

  - [Continuous Distribution](#continuous-distribution)
  - [Discrete Distribution](#discrete-distribution)
    - [Hypergeometric Distribution](#hypergeometric-distribution)
    - [Binomial Distribution](#binomial-distribution)
- [Monte Carlo Simulations](#monte-carlo-simulations) 
  - [Monte Carlo Fallacy](#monte-carlo-fallacy) 
  - [Crude Monte Carlo](#crude-monte-carlo)
  - [Law of Large Numbers](#law-of-large-numbers)
- [Methodology](#methodology)
   - [Notation Used](#notation-used)
   - [Crude Monte Carlo Simulation](#crude-monte-carlo-simulation)
     - [Programming the Simulation](#programming-the-simulation)
     - [Use of Random Numbers](#use-of-random-numbers)
     - [Computing Statistics](#computing-statistics)
   - [Statistical Calculation](#statistical-calculation)
     - [Binomial Distribution Formula](#binomial-distribution-formula)
     - [Hypergeometric Distribution Formula](#hypergeometric-distribution-formula)
     - [Summation Formula](#summation-formula) 
- [References](#references)
- [Appendices](#appendices)
   - [Appendix A: Definitions of Terms](#appendix-a-definitions-of-terms)
- [Contributors](#contributors) 

## Introduction

> 

Reviewer: Please ensure line widths of ~120 are adhered to for easier review comments and comparison of changes - for report in general.
_**Note:** This excludes links, tables and equations, which should not have newline characters in between._

> The paper serves as building blocks to aid in the answering of questions regarding the Tari Network. In order to perform analysis and draw accurate conclusions, a set of skills and programmes need to be built and honed. Among these, is an understanding of statistics. This report focuses on types of distribution and the deriving of their formulae and touches on the simulations, specifically the Monte Carlo Simulation. 

Reviewer: ```suggestion
The paper discusses building blocks to aid in answering of questions regarding the Tari Network. In order to perform analysis and draw accurate conclusions, a set of skills and programs need to be built and honed. Among these, is an understanding of statistics. This report focuses on types of distribution and the deriving of their formulae and touches on the simulations, specifically the Monte Carlo Simulation. 
```

## Types of Distribution 

There are 6 high-level types of distributions. These include discrete, continuous , mixed discrete/continuous, joint, non-numeric and miscellaneous. 

When considering solving the probability of an attacker controlling the majority of nodes in the network, the various 
types of probability distributions of the specific circumstances and variables of the 

> problem need to be analyzed. There are two categories of probability distribution: finite and infinite support [[1]]. 

Reviewer: Please add definitions for `Discrete distribution with finite support` and `Discrete distribution with infinite support` under `Appendix A: Definitions of Terms` and reference with superscript trick

### Continuous Distribution 

A continuous random variable is a random variable with an infinite and uncountable set and range of possible values [[11]].

Probabilities of continuous random variables (*X*) are defined as the area under the curve of its Probability Density 
Function<sup>[def][pdf~]</sup> (PDF). Therefore only ranges of values can have a nonzero probability. The probability that 

> a continuous random variable equals some value is always zero [[11]].

Reviewer: All these lines `<br />` can be removed. the only requirement is an open empty line before and after the html of the image

| Poisson Binomial  Distribution       | The number of successes in a series of independent Yes/No experiments with different success probabilities. |

A discrete distribution with infinite support can have an infinite number of possible realizations. Examples of discrete distribution with **infinite** support include the following: 

| Type of Infinite Discrete Distribution | Description [[1]]                                            |
| -------------------------------------- | ------------------------------------------------------------ |
| Boltzmann Distribution                 | A discrete distribution important in statistical physics, which describes the probabilities of the various discrete energy levels of a system in thermal equilibrium. It has a continuous analog. |
| Geometric Distribution                 | A discrete distribution that describes the number of attempts needed to get the first success in a series of independent Bernoulli trials or, alternatively, only the number of losses before the first success (i.e. one less). |

> | Negative Binomial/Pascal Distribution  | A generalization of the geometric distribution of the $nth$ success.             |```

Reviewer: ```suggestion
| Negative Binomial/Pascal Distribution  | A generalization of the geometric distribution of the $nth$ success.             |
```

### Law of Large Numbers 

The Law of Large Numbers (LLN), in probability and statistics, states that as a sample size grows, its mean gets closer 
to the average of the whole population. In statistical analysis, the LLN can be applied to a variety of 
subjects. It may not be feasible to poll every individual within a given population to collect the required amount of 
data. However, every additional data point gathered has the potential to increase the likelihood that the outcome is an 
accurate measure of the mean [[7]]. 

The LLN is crucial because it guarantees stable, long-term results for the averages of some random event [[8]]. 

<br />
<p align="center"><img src="assets/law_of_large_numbers.png" width="650" /></p>
<br />

The preceding figure illustrates the LLN using a particular run of rolls of a single dice. As can be seen in the figure, as the number of rolls in this run increases, the average of the values of all the results approaches 3.5. While different runs would show a different shape over a small number of throws (at the left), over a large number of rolls (to the right), they would be extremely similar [[9]].

# Methodology 

> ##Notation Used  

Reviewer: ```suggestion
## Notation Used
```

*bad nodes or more from a pool of total nodes?"* To this end, we perform a variable number of experiments and 
count how many times the threshold is met, in order to estimate the probability. 

### Use of Random Numbers  

Some problems in science and technology are described by "exact" mathematics, leading to "precise" results, e.g. 
throwing a ball and oscillating a system. Some problems appear physically uncertain, e.g. rolling a die and molecular motion. 
Random numbers can be used to mimic the uncertainty of the experiment. 

Random numbers make it possible to simulate physical systems with uncertainty, in input data or the process. 

### Computing Statistics 

The mean, median and mode are the arithmetic average of a data set. This is found by adding the enumbers in a data set and dividing by how many numbers there are. 

The mode is the value that occurs with most often in a data set.

The median is the middle number in a data set with the numbers are listed in either ascending or descending order.

<br />

<p align="center"><img src="assets/mode_median_mean.png" width="170" /></p>
<br />

> To describe a set of random numbers $xi$, we are often interested in two things:

Reviewer: `$xi$` still does not make sense. Is it `$x$` or $x_i$?
